src/engine/Server.cpp

```diff
@@ -912,25 +917,35 @@ json Server::processUpdateImpl(
 CPP_template_2(typename RequestT, typename ResponseT)(
     requires ad_utility::httpUtils::HttpRequest<RequestT>)
     Awaitable<void> Server::processUpdate(
-        ParsedQuery&& update, const ad_utility::Timer& requestTimer,
+        std::vector<ParsedQuery>&& updates,
+        const ad_utility::Timer& requestTimer,
         ad_utility::SharedCancellationHandle cancellationHandle,
         QueryExecutionContext& qec, const RequestT& request, ResponseT&& send,
         TimeLimit timeLimit) {
-  AD_CORRECTNESS_CHECK(update.hasUpdateClause());
-  PlannedQuery plannedQuery =
-      co_await planQuery(updateThreadPool_, std::move(update), requestTimer,
-                         timeLimit, qec, cancellationHandle);
+  AD_CORRECTNESS_CHECK(ql::ranges::all_of(
+      updates, [](const ParsedQuery& p) { return p.hasUpdateClause(); }));
+  std::vector<PlannedQuery> plannedUpdates;
+  for (ParsedQuery update : std::move(updates)) {
+    plannedUpdates.push_back(
+        co_await planQuery(updateThreadPool_, std::move(update), requestTimer,
+                           timeLimit, qec, cancellationHandle));
+  }
   auto coroutine = computeInNewThread(
       updateThreadPool_,
-      [this, &requestTimer, &cancellationHandle, &plannedQuery]() {
+      [this, &requestTimer, &cancellationHandle, &plannedUpdates]() {
         // Update the delta triples.
         return index_.deltaTriplesManager().modify<nlohmann::json>(
             [this, &requestTimer, &cancellationHandle,
-             &plannedQuery](auto& deltaTriples) {
-              // Use `this` explicitly to silence false-positive errors on
-              // captured `this` being unused.
-              return this->processUpdateImpl(plannedQuery, requestTimer,
-                                             cancellationHandle, deltaTriples);
+             &plannedUpdates](auto& deltaTriples) {
+              json results = json::array();
+              for (const auto& plannedUpdate : plannedUpdates) {
+                // Use `this` explicitly to silence false-positive errors on
+                // captured `this` being unused.
+                results.push_back(
+                    this->processUpdateImpl(plannedUpdate, requestTimer,
+                                            cancellationHandle, deltaTriples));
```

Comment:
I think we can do something more efficient here
(but probably in the future, at least add a TODO), 
For the multiple chained updates, we could inform the deltaTriples about chained updates, s.t. we only do the propagation to the located triples etc. once we are completely done.

Reply:
(Also add the TODO in the delta triples near the modify function, s.t. we might stumble upon it at some time).

Reply:
We still need the updates to be available during the evaluation of the where clause of the next update. So I'm unsure as to whether this is feasible.